src/hooks/useLocation.jsx

```diff
@@ -0,0 +1,20 @@
+import {useEffect, useState} from "react";
+
+export function useLocation() {
+    const [location, setLocation] = useState({});
+
+    useEffect(() => {
+        getLocation();
+    }, []);
+
+    function getLocation() {
+        navigator.geolocation.getCurrentPosition(position => {
```

Comment:
what do you do if the user declines permission for navigation?

Reply:
nice question)